InnerEye/ML/augmentations/augmentation_for_segmentation_utils.py

```diff
@@ -126,3 +128,45 @@ def random_crop(sample: Sample,
         metadata=sample.metadata
     )
     return sample, center
+
+
+class BasicAugmentations(Transform3D[Sample]):
+    """
+    Transform3D for basic augmentations on a SegmentationSample
+    """
+
+    def __call__(self, sample: Sample) -> Sample:
+        image, labels = self.transform(
+            image=sample.image,
+            labels=sample.labels)
+        return sample.clone_with_overrides(
+            image=image,
+            labels=labels,
+        )
+
+    @staticmethod
+    def transform(image: Union[np.ndarray, torch.Tensor],
+                  labels: Union[np.ndarray, torch.Tensor]) -> \
+            Tuple[np.ndarray, np.ndarray]:
+        """
+        Applies 1 transform with 0.5 probability
+        :param image: channels x image dimensions
+        :param labels: channels x image dimensions
+        :return: A tuple of image channels transformed and labels transformed
+        """
+        import torchio as tio
+        subject = tio.Subject(
+            image=tio.ScalarImage(tensor=image),
+            labels=tio.LabelMap(tensor=labels)
+        )
+        augment = tio.Compose([
+            tio.OneOf({
+                tio.RandomAffine(
+                    degrees=20),
+                tio.RandomNoise(),
+                tio.RandomMotion(),
+                tio.RandomBlur(),
+            }, p=0.5)
+        ])
+        transformed_subject = augment(subject)
+        return transformed_subject.image.numpy(), transformed_subject.labels.numpy()
```

Comment:
If you're applying "rotate", the labels also need to be rotated, is that the case now?

Reply:
yes the torchio subject deals with this.